vmaas_sync/vmaas_sync.py

```diff
@@ -190,13 +190,13 @@ def re_evaluate_systems(self):
         """Schedule re-evaluation for all systems in DB."""
         LOGGER.info("Re-evaluating all systems")
         conn = DatabaseHandler.get_connection()
-        cur = conn.cursor()
-        cur.execute("select inventory_id from system_platform")
-        # reevaluate updates for every system in the DB
-        for inventory_id, in cur.fetchall():
-            self.evaluator_queue.send({"type": "re-evaluate_system", "system_id": inventory_id})
-        cur.close()
+        with NamedCursor(conn) as cur:
+            cur.execute("select inventory_id from system_platform")
+            # reevaluate updates for every system in the DB
+            for inventory_id, in cur.fetchall():
```

Comment:
when using named cursor don't use fetchall, it doesn't solve the issue, cursor object is iterable and it fetches rows in batches

http://initd.org/psycopg/docs/cursor.html

```
iterating over a named cursor fetches itersize records at time from the backend. Previously only one record was fetched per roundtrip, resulting in a large overhead.
```

http://initd.org/psycopg/docs/cursor.html#cursor.itersize

Reply:
This should now be fixed.